v2/googlecloud-to-googlecloud/docs/SpannerToBigQuery/README.md

```diff
@@ -0,0 +1,184 @@
+batch pipeline that reads data from a Spanner Table and writes them to a BigQuery table.
+
+##Getting Started
+
+### Requirements
+* Java 8
+* Maven
+* The Spanner table exists
+* The BigQuery Output table exists
+
+
+###Building Template
+This is a Flex Template meaning that the pipeline code will be containerized and the container will be run on Dataflow.
+
+####Building Container Image
+* Set environment variables to be used in the build process.
+
+```sh
+export PROJECT=<project-id>
+export IMAGE_NAME=googlecloud-to-googlecloud
+export BUCKET=gs://<GCS-bucket-name>
+export TEMPLATE_MODULE=googlecloud-to-googlecloud
+export COMMAND_MODULE=spanner-to-bigquery
+
+export TARGET_GCR_IMAGE="gcr.io/$PROJECT/images/$IMAGE_NAME"
+export BASE_CONTAINER_IMAGE=gcr.io/dataflow-templates-base/java8-template-launcher-base
+export BASE_CONTAINER_IMAGE_VERSION=latest
+export APP_ROOT="/template/$COMMAND_MODULE"
+export COMMAND_SPEC="$APP_ROOT/resources/$COMMAND_MODULE-command-spec.json"
+
+export IMAGE_SPEC_LOCAL_PATH=<path to spanner-to-bigquery-image-spec.json>
+export IMAGE_SPEC_GCSPATH="$BUCKET/images/$COMMAND_MODULE-image-spec.json"
+```
+* Build and push image to Google Container Repository from the v2 directory.
+```sh
+mvn clean package -Dimage=$TARGET_GCR_IMAGE \
+                  -Dbase-container-image=$BASE_CONTAINER_IMAGE \
+                  -Dbase-container-image.version=$BASE_CONTAINER_IMAGE_VERSION \
+                  -Dapp-root=$APP_ROOT \
+                  -Dcommand-spec=$COMMAND_SPEC
+                  -am -pl $TEMPLATE_MODULE
+```
+
+#### Creating Image Spec
+* Create spanner-to-bigquery-image-spec.json file locally.
+```sh
+{
+    "name": "Spanner to BigQuery",
+    "description": "Reads records from Spanner table and writes them to a BigQuery table",
+    "parameters": [{
+      "name": "spannerTable",
+      "label": "Spanner Table ID",
+      "helpText": "The Spanner table to read records from.",
+      "isOptional": false,
+      "paramType": "TEXT"
+    },
+      {
+        "name": "spannerInstanceId",
+        "label": "Spanner Instance ID",
+        "helpText": "The Spanner instance for the Spanner table.",
+        "isOptional": false,
+        "paramType": "TEXT"
+      },
+      {
+        "name": "spannerDatabaseId",
+        "label": "Spanner Database ID",
+        "helpText": "The Spanner database containing the Spanner table.",
+        "isOptional": false,
+        "paramType": "TEXT"
+      },
+      {
+        "name": "spannerRpcPriority",
+        "label": "Spanner RPC Priority",
+        "helpText": "The priority of Spanner job. Must be one of the following: [HIGH, MEDIUM, LOW]. Default is HIGH.",
+        "isOptional": true,
+        "regexes": [
+        "^(HIGH|MEDIUM|LOW)$"
+      ],
+        "paramType": "TEXT"
+      },
+      {
+        "name": "sqlQuery",
+        "label": "SQL Query",
+        "helpText": "The SQL Query used to read data from the Spanner table.",
+        "isOptional": false,
```

Comment:
Isn't the SQL query required?

Reply:
Setting "isOptional" to false implies it's compulsory. Or should i leave out the "isOptional" field since I am setting it to false anyway?